packages/ethers5/src/client.ts

```diff
@@ -168,8 +171,39 @@ export class Web3Modal extends Web3ModalScaffold {
         WalletConnectProvider.on('display_uri', (uri: string) => {
           onUri(uri)
         })
-
-        await WalletConnectProvider.connect()
+        console.log('SIWEController', SIWEController, siweConfig)
+        console.log('WalletConnectProvider init opts', options)
+
+        if (siweConfig?.options?.enabled) {
+          const result = await WalletConnectProvider.authenticate({
+            nonce: await siweConfig.getNonce(),
+            methods: OPTIONAL_METHODS,
+            ...siweConfig.options.messageParams
+          })
+          // Auths is an array of signed CACAO objects https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-74.md
+          if (result?.auths?.[0]) {
+            const { p, s } = result.auths[0]
```

Comment:
Please use a variable for better readability. Btw don't we need to iterate over the signed CACAOs and do this for each of them ?
```suggestion
          const signedCacaos = result?.auths?.[0]
          if (signedCacaos) {
            const { p, s } = signedCacaos
```